### PR TITLE
Add in new tesState for preempted state.

### LINKED ISSUE #183: Proposal: New TES State for Preemption
Problem:
Cloud platform support preemption of machines. Cromwell needs a way to understand that state to implement the correct logic of dealing with task resubmission

Solution
Add in new state to tesState: "Preempted"

Linked to this issue: https://github.com/microsoft/CromwellOnAzure/issues/451

### PR DESCRIPTION
**Pull request by vsmalladi**
Close #183 

Clouds/Job scheduler have preempted rules:

* [GCP](https://cloud.google.com/compute/docs/instances/preemptible)
* [Slurm](https://slurm.schedmd.com/slurm.conf.html#PreemptMode)
* [Azure](https://docs.microsoft.com/en-us/azure/virtual-machines/spot-vms)
* [AWS](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/how-spot-instances-work.html)

Solves : https://github.com/microsoft/CromwellOnAzure/issues/451

### CODE REVIEW

File: openapi/task_execution_service.openapi.yaml [hunk 1]
```diff
@@ -554,7 +554,7 @@ components:
         backend_parameters_strict:
             type: boolean
             description: |-
-                If set to true, backends should fail the task if any backend_parameters
```
Comment (MattMcL4475) [inline openapi/task_execution_service.openapi.yaml:557]: Revert this change as its unrelated

File: openapi/task_execution_service.openapi.yaml [hunk 2]
```diff
@@ -623,6 +623,9 @@ components:
         for example an upload failed due to network issues, the worker's ran out
         of disk space, etc.
          - `CANCELED`: The task was canceled by the user.
+         - `PREEMTED`: The task is stopped (preempted) by the system. The reasons for this would be tied to
```
Comment (MattMcL4475) [inline openapi/task_execution_service.openapi.yaml:626]: Should be PREEMPTED

File: openapi/task_execution_service.openapi.yaml [hunk 3]
```diff
@@ -635,6 +638,7 @@ components:
       - EXECUTOR_ERROR
       - SYSTEM_ERROR
       - CANCELED
+      - PREEMTED
```
Comment (MattMcL4475) [inline openapi/task_execution_service.openapi.yaml:641]: PREEMPTED

File: openapi/task_execution_service.openapi.yaml (hunk 1 above)
Comment (vsmalladi) [reply openapi/task_execution_service.openapi.yaml]: Fixed

File: openapi/task_execution_service.openapi.yaml [hunk 4]
```diff
@@ -589,7 +589,7 @@ components:
           tesResources_backend_parameters:
             type: array
             description: |-
-              Lists all tesResources.backend_parameters keys supported 
+              Lists all tesResources.backend_parameters keys supported
```
[the remark this answers is not in the document]
Comment (vsmalladi) [reply openapi/task_execution_service.openapi.yaml]: Fixed

File: openapi/task_execution_service.openapi.yaml (hunk 3 above)
Comment (vsmalladi) [reply openapi/task_execution_service.openapi.yaml]: Fixed

**Review by MattMcL4475 — APPROVED** [review]

**Review by kellrott — APPROVED** [review]
